mmv1/third_party/terraform/acctest/vcr_utils.go

```diff
@@ -243,13 +282,35 @@ func initializeReleaseDiffTest(c resource.TestCase, testName string) resource.Te
 		}
 	}
 
+	c.Steps = InsertDiffSteps(c, tempFile, releaseProvider, localProviderName)
+
+	return c
+}
+
+// InsertDiffSteps inserts a new step into the test case that reformats the config to use the release provider - this allows us to see the diff
+// between the local provider and the release provider. for a certain test, this will be used to see the diff between the local provider and the release provider.
+func InsertDiffSteps(c resource.TestCase, tempFile *os.File, releaseProvider string, localProviderName string) []resource.TestStep {
+	var countSteps = 0
+
 	var replacementSteps []resource.TestStep
 	for _, testStep := range c.Steps {
+		// todo: add preconfig - categorize test failures (add flag to steps that if they fail is a diff failure)
 		if testStep.Config != "" {
 			ogConfig := testStep.Config
+			fmt.Fprintf(tempFile, "Original config: %s\n", ogConfig)
+			fmt.Fprintf(tempFile, "Reformatting config to use provider: %s\n", localProviderName)
 			testStep.Config = ReformConfigWithProvider(ogConfig, localProviderName)
-			if testStep.ExpectError == nil && testStep.PlanOnly == false {
+			fmt.Fprintf(tempFile, "Reformatted config: %s\n", testStep.Config)
+			testStep.PreConfig = func() {
+				fmt.Fprintf(tempFile, "[Diff] Step %d\n", countSteps)
+				countSteps++
```

Comment:
I think you may want this incremented as the first line under the for loop, but could you confirm the behavior? I think this will basically count each of the new steps, which seems like it would be of limited use. If if was the count of the step in the original test, then I think it would be more useful as a lookup of sorts for someone to go back to the test source code and look for where the problem occurred.